### PR TITLE
Validate links in md files

### PR DESCRIPTION
**Pull request by mi-hol**
@karniv00l please consider this is my first js script :)

Test results:
```

PS C:\Users\mholzer\Documents\git\fome-wiki\scripts> node linkValidator.js .\linkValidator.test.md
Validating rules for URL links in:  ./linkValidator.test.md
❌ Failed

Number of Errors found: 15
[./linkValidator.test.md][Line:5][ErrType:MdLink]"[T] (https://wikiXfome.tech/01-test.md)"
[./linkValidator.test.md][Line:5][ErrType:NumPrefix]"[T] (https://wikiXfome.tech/01-test.md)"
[./linkValidator.test.md][Line:9][ErrType:MdLink]"blah (https://wiki.fome.tech/01-test.md)"
[./linkValidator.test.md][Line:9][ErrType:NumPrefix]"blah (https://wiki.fome.tech/01-test.md)"
[./linkValidator.test.md][Line:13][ErrType:AbsLink]"[T](https://wiki.fome.tech/01-test.md)"
[./linkValidator.test.md][Line:13][ErrType:MdLink]"[T](https://wiki.fome.tech/01-test.md)"
[./linkValidator.test.md][Line:13][ErrType:NumPrefix]"[T](https://wiki.fome.tech/01-test.md)"
[./linkValidator.test.md][Line:17][ErrType:AbsLink]"[T](https://wikiXfome.tech/test) # note: false positive, but extreme unlikely to hit in real world"
[./linkValidator.test.md][Line:21][ErrType:MdLink]"(/01-test.md)"
[./linkValidator.test.md][Line:21][ErrType:NumPrefix]"(/01-test.md)"
[./linkValidator.test.md][Line:25][ErrType:MdLink]"(test.md)"
[./linkValidator.test.md][Line:29][ErrType:MdLink]"(test.md) blah"
[./linkValidator.test.md][Line:33][ErrType:MdLink]"blah (test.md) blah"
[./linkValidator.test.md][Line:37][ErrType:MdLink]"(../test.mdx)"
[./linkValidator.test.md][Line:49][ErrType:NumPrefix]"(/01-test)"
PS C:\Users\mholzer\Documents\git\fome-wiki\scripts> node .\linkValidator.js '..\docs\**\*.md?(x)'
Validating rules for URL links in:  144 files
✅ Ok

```

### CODE REVIEW

File: scripts/linkValidator.js [hunk 1]
```diff
@@ -1,59 +1,107 @@
...
-  }
 }
 
 /**
  * Load all md and mdx files from / docs and process them
- */
+*/
 const main = () => {
-  const files = glob.sync('docs/**/*.md?(x)');
+  // ToDo: change in package.json "lint:links": "node scripts/linkValidator.js 'docs/**/*.md?(x)'",
```
Comment (karniv00l) [inline scripts/linkValidator.js:81]: no need for TODO, you can add it in this PR

File: scripts/linkValidator.test.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Tests for linkValidator.js
```
Comment (karniv00l) [inline scripts/linkValidator.test.md:1]: maybe consider moving this file to `/scripts/tests/`?

File: scripts/linkValidator.js (hunk 1 above)
Comment (mi-hol) [reply scripts/linkValidator.js:81]: thats what I forget in initial PR :(

File: scripts/linkValidator.test.md (hunk 2 above)
Comment (mi-hol) [reply scripts/linkValidator.test.md:1]: great suggestion, done

**Comment by mi-hol** [comment]
node .\linkValidator.js '..\docs\**\*.doc'
C:\Users\mholzer\Documents\git\fome-wiki\scripts\linkValidator.js:26
   throw new Error("UsageError: no files qualify!")
   ^

Error: UsageError: no files qualify!
    at validateDocRules (C:\Users\mholzer\Documents\git\fome-wiki\scripts\linkValidator.js:26:10)
    at main (C:\Users\mholzer\Documents\git\fome-wiki\scripts\linkValidator.js:86:3)
    at Object.<anonymous> (C:\Users\mholzer\Documents\git\fome-wiki\scripts\linkValidator.js:105:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47

Node.js v20.6.1

**Pull request merged**